On share logic

jeremyjpenafiel opened the pull request.

jeremyjpenafiel committed:
implemented onUnlikeLogic and unit tests

jeremyjpenafiel committed:
fixed formatting

jeremyjpenafiel committed:
implemented onShareLogic and unit tests

jeremyjpenafiel committed:
added the correct src path for user timeline

jeremyjpenafiel committed:
Update share-logics.test.ts

JPvenadas commented on functions/src/business-logics/share-logics.ts:
```diff
@@ -1,4 +1,80 @@
+    } = post;
+    const timelineLogicResultDoc: LogicResultDoc = {
+      action: "copy",
+      dstPath: `users/${user["@id"]}/timeline/${postId}`,
```
minor suggestion: instead of using users["@id"] you can actually use uid, which you can also get from event context.

JPvenadas commented on functions/src/business-logics/share-logics.ts:
```diff
@@ -1,4 +1,80 @@
+
+    const notificationLogicResultDoc: LogicResultDoc = {
+      action: "create",
+      dstPath: `users/${postAuthorId}/notifications/${postId}`,
```
missing doc, we cannot use "create" action type without a document

JPvenadas commented on functions/src/business-logics/share-logics.ts:
```diff
@@ -1,4 +1,80 @@
+    };
+    const postLogicResultDoc: LogicResultDoc = {
+      action: "create",
+      dstPath: docPath,
```
the value of docPath here is "posts/postId/shares/shareId" because you are using the share entity not post. if you want to use the post path you can get it in the postDocRef

JPvenadas commented on functions/src/business-logics/share-logics.ts:
```diff
@@ -1,4 +1,80 @@
 //  create a notification for post's author
+
+
+export const onShareLogic: LogicConfig = {
```
Additional task: create another logic result doc that will add the userview of the user in the "posts/postId/share/[userId]". you can expect the docpath to have this path so you can just use it.

jeremyjpenafiel committed:
Merge branch 'main' into onUnlikeLogic

jeremyjpenafiel committed:
fixed import errors

jeremyjpenafiel committed:
implemented required changes

- used uid instead of user["@id"]
- created a notificationDoc for notificationLogicResultDoc
- changed to appropriate path to get post
- created another logicResultDoc for adding the userView

jeremyjpenafiel commented:
Implemented changes
- used uid instead of user["@id"]
- created a notificationDoc for notificationLogicResultDoc
- changed to appropriate path to update post in postLogicResultDoc
- created another logicResultDoc for adding the userView

jeremyjpenafiel committed:
removed comments

jeremyjpenafiel committed:
used userView instead of user in userLogicResultDoc

jeremyjpenafiel committed:
updated tests and notification

- updated tests for corresponding changes to onShareLogic

- update notification id to be the same as the eventContext's docId

jeremyjpenafiel committed:
Merge branch 'main' into onShareLogic

jeremyjpenafiel committed:
Merge branch 'onUnlikeLogic' into onShareLogic

jeremyjpenafiel commented:
- used userView instead of user in userLogicResultDoc
- updated tests for corresponding changes to onShareLogic
- updated notification id to be the same as eventContext's docId
- merged main and branch onUnlikeLogic
